cli2telegram/util.py

```diff
@@ -38,28 +39,33 @@ def send_message(bot: Bot, chat_id: str, message: str, parse_mode: str = None, r
     """
     from emoji import emojize
     emojized_text = emojize(message, use_aliases=True)
-    return bot.send_message(chat_id=chat_id, parse_mode=parse_mode, text=emojized_text, reply_to_message_id=reply_to,
-                            reply_markup=menu, timeout=10)
+    return bot.send_message(
+        chat_id=chat_id, parse_mode=parse_mode, text=emojized_text, reply_to_message_id=reply_to, reply_markup=menu, timeout=10
+    )
 
-
-def prepare_code_message(lines: [str]) -> str:
+def split_message(lines: [str]) -> [str]:
     """
-    Prepares the given lines of text to send them as a code block message
+    Split lines into multiple messages if the maximum number of character is exceeded.
     :param lines: text lines
-    :return: prepared message
+    :return: list of messages
     """
-    lines = list(map(lambda x: x + "\n" if not x.endswith("\n") else x, lines))
```

Comment:
I don't think you can just remove this. This code is supposed to "normalize" newlines in a way that the telegram API understands (which is only `\n`). When entering messages on the CLI f.ex., there might be newlines in it, which are not "written out" as `\n`. The test cases for this are very sparse atm and they don't seem to tun on PRs. I will investigate this.

Reply:
Basically these three variants:

```python
"""
a
b
c
"""
```

```python
"a\nb\nc"
```

```python
"""
a\n
b\n
c
"""
```

should all result in a message that looks like this:

```
a
b
c
```

Reply:
Combining this with the code block adds some complexity to this.

Reply:
I think that should be the expected behavior as \n is read as two characters and not as the newline character that would display the subsequent text on a new line

Reply:
Sorry, but I dont understand what you mean?

Reply:
If I write "some\nmessage" I think it makes sense that it is interpreted literally otherwise I'd write "some
message" with the actual newline

Reply:
Outside of python that is...
To get the effect you're saying it's enough to apply `.replace("\\n", "\n")` to the joined lines at row 52 in util.py

Reply:
Ah, well, yes basically I agree. However, we would need to escape literal `\n` before sending them to telegram, because telegram will interpret `\n` as a newline. I am not sure if escaping them is possible, also I had issues with getting newlines to work correctly when passing them from the CLI. At least thats what I remember... I wrote this code ages ago :sweat_smile:

So, for that reason, I think it would be easier to keep the initial behavior for this PR, but I am open to discussing how we can improve it in a new issue. I think an expansion of test scenarios would be the first step and would also help a lot to ensure compatibility in the future.

Reply:
I don't think that telegram does what you're saying. If I send the first example message without `.replace("\\n", "\n")`:
```
"""
a
b
c
"""
```
I get:
```
a
b
c
```
Second example:
```
"a\nb\nc"
```
I get:
```
a\nb\nc
```

Third example:
```
"""
a\n
b\n
c
"""
```
Result:
```
a\n
b\n
c
```

Reply:
Thx, I will need to to some investigation myself.
Not sure when I will get to it but I will try asap.

Reply:
Thx, I will need to to some investigation myself.
Not sure when I will get to it but I will try asap.

Reply:
That's ok no rush

Reply:
Ok so I did some basic testing and I can't figure out anymore what the reason for this code was...
I think it had to do with terminal stdin and how "newlines" are handled.
So I will accept this for now, add some tests to make sure it doesn't change in the future, and fix anything that might have broken later on.